app/src/main/java/com/pvp/app/ui/screen/layout/RewardDialog.kt

```diff
@@ -0,0 +1,160 @@
+package com.pvp.app.ui.screen.layout
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Star
+import androidx.compose.material.icons.outlined.Stars
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.pvp.app.model.Decoration
+import com.pvp.app.model.Reward
+import com.pvp.app.model.Streak
+import com.pvp.app.ui.common.AsyncImage
+import com.pvp.app.ui.common.Button
+import com.pvp.app.ui.common.lighten
+
+@Composable
+private fun DecorationCard(decoration: Decoration) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(8.dp),
+        verticalArrangement = Arrangement.Center
+    ) {
+        Text(
+            modifier = Modifier.padding(8.dp),
+            style = MaterialTheme.typography.titleMedium,
+            text = "${decoration.name} decoration!"
+        )
+
+        AsyncImage(
+            contentDescription = "Decoration ${decoration.name} image",
+            modifier = Modifier
+                .size(96.dp)
+                .clip(MaterialTheme.shapes.extraSmall)
+                .background(
+                    color = MaterialTheme.colorScheme.inverseOnSurface.lighten(),
+                    shape = MaterialTheme.shapes.extraSmall
+                ),
+            url = decoration.imageRepresentativeUrl
+        )
+    }
+}
+
+@Composable
+fun RewardDialog(
+    isOpen: Boolean,
+    onClose: () -> Unit,
+    reward: Reward,
+    streak: Streak
+) {
+    if (!isOpen) {
+        return
+    }
+
+    Dialog(onDismissRequest = onClose) {
+        Surface(
```

Comment:
This looks redundant, can move those params into inner Column container. Probably Surface's `color` is backgroundColor, so inner column already overrides it with its modifier.

Reply:
Surface is necessary to make the dialog have rounded corners. Color can be removed though.

Reply:
But I mean, Column can already have both, so Surface is just another redundant nesting layer, no?
```kotlin
modifier = Modifier
    .clip(MaterialTheme.shapes.medium) // instead of RoundedCornerShape(16.dp)
    .background(MaterialTheme.colorScheme.surface) // or the one Column already uses, since it overrides Surface's color 🤷
```

Reply:
Oh ffs I forgot you need to clip before setting the background, that's why it didn't work on the column before..